File: calculations/calc.cpp
int mMSD  (int M) {
    // (8!)^36
    int i, poss=1;
    for (i=1; i<=M; i++) {
        poss = ((Factorial(i)) ^ 36);
    }
}


int Factorial (int M) {
	int i, factorial=1;

	for (i=1; i<=M; i++) {
		factorial = factorial * i;
	}
	return factorial;
}

File: calculations/calc.hs
factorial :: Int -> Int
factorial 0 = 1
factorial n = n * factorial (n - 1)

-- This function breaks GHCi--
minimsd :: Int -> Int
minimsd 0 = 1
minimsd n = ( ( n * ( factorial ( n - 1 ) ) ) ^ 36 ) // Broken in ghci due to stack limit

-- This function should not break GHCi --
mmsdtail :: Int -> Int
mmsdtail n = 
